esmvaltool/preprocessor/_time.py

```diff
@@ -162,10 +171,68 @@ def seasonal_mean(cube):
     cube = cube.aggregated_by(['clim_season', 'season_year'],
                               iris.analysis.MEAN)
 
-    # TODO: This preprocessor is not calendar independent.
+    # this func returns an approximation to 3 months
     def spans_three_months(time):
-        """Check for three months"""
+        """Check for three months."""
         return (time.bound[1] - time.bound[0]) == 2160
 
     three_months_bound = iris.Constraint(time=spans_three_months)
     return cube.extract(three_months_bound)
+
+
+def _align_time_axes(cubes):
+    """
+    Align time axis for cubes so they can be subtracted.
+
+    Operations on time units, calendars, time points and auxiliary
+    coordinates so that any cube from cubes can be subtracted from any
+    other cube from cubes.
+
+    Arguments
+    ---------
+        cubes: list of iris.cube.Cube
+
+    Returns
+    -------
+    list of iris.cube.Cube instances
+    """
+    for cube in cubes:
+        # fix units; leave calendars
+        # this is not normally needed unless working
+        # outside the CMOR standards
+        cube.coord('time').convert_units(
+            cf_units.Unit('days since 1950-1-1 00:00:00',
+                          calendar=cube.coord('time').units.calendar))
+
+        # fix calendars
+        cube.coord('time').units = \
+            cf_units.Unit(cube.coord('time').units.origin,
+                          calendar='gregorian')
+
+        # standardize time points
+        time_c = [cell.point for cell in cube.coord('time').cells()]
+        cube.coord('time').cells = [
+            datetime.datetime(t.year, t.month, 15, 0, 0, 0) for t in time_c
```

Comment:
It looks like this only works for monthly data, what about daily, hourly etc?

Reply:
Also, wouldn't a choice of regridding schemes, like for the horizontal and vertical regridders, make more sense?

Reply:
mmyeahh...I added a few more cases kwarged on `frequency`; not sure about other types of data since I haven't yet worked with them; no man, no need for time-regridding, that would be overly complicated, all the cubes need are common points, same aux coords and the same bounds

Reply:
But that would be inaccurate, right? @mattiarighi @ledm Do you know if it would make sense to support proper temporal regridding? I think this could be a useful feature for preprocessing data going into computing multimodel statistics too.

Reply:
As we discussed in another PR, I am a bit skeptical towards temporal regridding.
For monthly mean data that's easy, since the number of timesteps is always the same (12 per year).

But for higher time frequency it is calendar-dependent and we need to make assumptions on how to set a common time coordinate for all datasets when calendars are differing.

I also think that a non-standard calendar (like 360) is not suitable for temporally colocating model and observations on a daily or hourly basis, but just monthly.

Reply:
> For monthly mean data that's easy, since the number of timesteps is always the same (12 per year).

Yes, but I could image that it makes a big difference if the time point is the first of the month vs the middle or even last day of the month, right? Wouldn't it make sense to use linear interpolation for that case, if you want to compare data from the 1st of the month to the 15th?

Reply:
currently the function sets it to the 15th of the month as common point. Why would it make a difference if it is the first or the middle of the month? Issues appear if it's 29th of the month onwards because February